examples/counter/shared/src/lib.rs

```diff
@@ -0,0 +1,53 @@
+pub mod app;
+
+use lazy_static::lazy_static;
+use serde::{Deserialize, Serialize};
+use wasm_bindgen::prelude::wasm_bindgen;
+
+pub use crux_core::Request;
+use crux_core::{capability::CapabilityContext, render::Render, Core};
+pub use crux_http as http;
+use http::{Http, HttpRequest};
+
+pub use app::*;
+
+// TODO hide this plumbing
+
+uniffi_macros::include_scaffolding!("shared");
+
+lazy_static! {
+    static ref CORE: Core<Effect, App> = Core::new::<Capabilities>();
+}
+
+#[wasm_bindgen]
+pub fn message(data: &[u8]) -> Vec<u8> {
+    CORE.message(data)
+}
+
+#[wasm_bindgen]
+pub fn response(uuid: &[u8], data: &[u8]) -> Vec<u8> {
+    CORE.response(uuid, data)
+}
+
+#[wasm_bindgen]
+pub fn view() -> Vec<u8> {
+    CORE.view()
+}
+
+// TODO macro effect generation:
+// crux_macros::generate_effect!(Effect, MyCapabilities);
```

Comment:
An idea I had for this was to use [typetag](https://github.com/dtolnay/typetag).  Suspect we could set it up to serlalize a `dyn Operation` and get rid of the `Effect` enum altogether.

Reply:
Of course I just looked into it a tiny bit, and _suspect_ it won't work on wasm.  So never mind 😢

Reply:
Hmm. That would've been great. I got quite excited for a moment. We can write a macro to hide it

Reply:
Would that lose us the exhaustive pattern matching / switch on the shell side? It would have no way of knowing what the full list of tags is that it needs to handle, wouldn't it?

Reply:
Pretty sure that would have been solvable.  With `serde_generate` I'm 90% sure it'd just work, with `go-away` I could implement a similar thing to what typetag does under the hood.